src/translate.rs

```diff
@@ -184,6 +184,40 @@ pub struct StashMut<'a, P: Copy, T: ?Sized>(pub P, pub <T as ToGlibPtrMut<'a, P>
 where
     T: ToGlibPtrMut<'a, P>;
 
+#[derive(Debug)]
+pub struct Borrowed<T>(Option<T>);
+
+impl<T> Borrowed<T> {
+    pub fn new(val: T) -> Self {
+        Self(Some(val))
+    }
+
+    pub unsafe fn into_inner(mut self) -> T {
```

Comment:
This IMHO need comment about limits of usage this function

Reply:
Or maybe better instead add `from_borrowed` constructor?

Reply:
How would `from_borrowed()` work? I was thinking of that but it didn't seem like an improvement with what I came up with :)

Docs will definitely come if the approach is considered ok

Reply:
I think about something like `fn from_borrowed<E, F: FnOnce(E) -> T>(other: Borrowed<E>, f:F)-> Self`
With call
```
                $crate::translate::Borrowed::from_borrowed(
$crate::translate::from_glib_borrow::<_, $crate::boxed::Boxed<_, _>>(ptr),
          |v|   $name( v )
                )
```
If this don't work the :+1: for `into_inner`

Reply:
or we can just add `fn map<E, F: FnOnce(T) -> E>(self, f:F)-> Borrowed<E>`

Reply:
That's all doable but does not really improve the code and also does not make it any safer :) You still need to ensure that the value passed to the closure is not dropped anywhere.

I'd keep it as is.